docs/tools/doc_generator.py

```diff
@@ -0,0 +1,106 @@
+import ast
+from pathlib import Path
+from typing import Any, Dict, List, Set
+
+
+# TODO - API
+class DocGenerator:
+    def __init__(self, filename: Path) -> None:
+        assert str(filename)[-3:] == ".py"
+        self._filename = filename
+
+
+class ClassVisitor(ast.NodeVisitor):
+    def __init__(self) -> None:
+        self.classes: Set[Any] = set()
+
+    def visit_ClassDef(self, node: ast.ClassDef) -> None:
```

Comment:
You have reminded me we need pep8-naming as deps (totally missing). Would you kindly add that and fix this, please?

Reply:
select = [...],N                                                                                                                                 
on .flake8 needed as well

Reply:
Sure, will do.

Reply:
Oh I forgot to say that the function is overriding the super class in `ast`, so the name cannot be changed according to the pep8-naming, unfortunately.

Reply:
Alright, just add "noqa" exceptions on those entries and justify with a comment, please. Also squash the commits which change case back and forth with these exception comments.

Reply:
Roger that.